src/dbnode/storage/namespace_readers.go

```diff
@@ -238,6 +265,25 @@ func (m *namespaceReaderManager) get(
 	blockStart time.Time,
 	position readerPosition,
 ) (fs.DataFileSetReader, error) {
+	latestVolume, err := m.latestVolume(shard, blockStart)
+	if err != nil {
+		return nil, err
+	}
+
+	// If requesting an outdated volume, we need to start reading again from
+	// the beginning of the latest volume. The caller knows how to handle
+	// duplicate metadata, so doing this is okay.
+	//
+	// The previously cached reader for the outdated volume will eventually be
+	// cleaned up either during the ticking process or the next time
+	// UpdateOpenLease gets called, so we don't need to worry about closing it
+	// here.
```

Comment:
Hm interesting, not sure how I feel about this strategy long term.  This might be fine for now, but for any clusters that regularly get cold writes to most shards/block starts, it might make it impossible to ever finish a bootstrap as it just gets continually reset.